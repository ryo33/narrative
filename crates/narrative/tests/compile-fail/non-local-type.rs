// This type is defined outside the story trait
struct NotLocalType;

#[narrative::story("Test non-local type error")]
trait TestStory {
    #[step("Given a non-local type")]
    fn use_non_local_type(&self, arg: NotLocalType);
}

fn main() {}