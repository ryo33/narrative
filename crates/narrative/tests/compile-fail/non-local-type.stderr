error: macros that expand to items must be delimited with braces or followed by a semicolon
 --> tests/compile-fail/non-local-type.rs:7:34
  |
7 |     fn use_non_local_type(&self, arg: NotLocalType);
  |                                  ^^^
  |
help: change the delimiters to curly braces
  |
7 -     fn use_non_local_type(&self, arg: NotLocalType);
7 +     fn use_non_local_type(&self, {}: NotLocalType);
  |
help: add a semicolon
  |
7 |     fn use_non_local_type(&self, arg;: NotLocalType);
  |                                     +

error: No attr arg or assignment found
 --> tests/compile-fail/non-local-type.rs:7:34
  |
7 |     fn use_non_local_type(&self, arg: NotLocalType);
  |                                  ^^^

error[E0412]: cannot find type `NotLocalType` in this scope
 --> tests/compile-fail/non-local-type.rs:7:39
  |
7 |     fn use_non_local_type(&self, arg: NotLocalType);
  |                                       ^^^^^^^^^^^^ not found in this scope
  |
  = help: consider importing this struct:
          crate::NotLocalType

error[E0425]: cannot find value `arg` in this scope
 --> tests/compile-fail/non-local-type.rs:7:34
  |
7 |     fn use_non_local_type(&self, arg: NotLocalType);
  |                                  ^^^ not found in this scope
  |
  = help: consider importing one of these items:
          crate::mod_TestStory::arg_values::use_non_local_type::arg
          crate::mod_TestStory::args::use_non_local_type::arg

error[E0599]: no method named `value` found for enum `mod_TestStory::args::use_non_local_type` in the current scope
 --> tests/compile-fail/non-local-type.rs:4:1
  |
4 | #[narrative::story("Test non-local type error")]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  | |
  | method not found in `mod_TestStory::args::use_non_local_type`
  | method `value` not found for this enum
  |
  = help: items from traits can only be used if the trait is implemented and in scope
  = note: the following traits define an item `value`, perhaps you need to implement one of them:
          candidate #1: `narrative::step::StepArg`
          candidate #2: `narrative::story::StoryConst`
  = note: this error originates in the attribute macro `narrative::story` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no method named `name` found for enum `mod_TestStory::args::use_non_local_type` in the current scope
 --> tests/compile-fail/non-local-type.rs:4:1
  |
4 | #[narrative::story("Test non-local type error")]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  | |
  | method not found in `mod_TestStory::args::use_non_local_type`
  | method `name` not found for this enum
  |
  = help: items from traits can only be used if the trait is implemented and in scope
  = note: the following traits define an item `name`, perhaps you need to implement one of them:
          candidate #1: `narrative::step::StepArg`
          candidate #2: `narrative::story::StoryConst`
  = note: this error originates in the attribute macro `narrative::story` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no method named `ty` found for enum `mod_TestStory::args::use_non_local_type` in the current scope
 --> tests/compile-fail/non-local-type.rs:4:1
  |
4 | #[narrative::story("Test non-local type error")]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  | |
  | method not found in `mod_TestStory::args::use_non_local_type`
  | method `ty` not found for this enum
  |
  = help: items from traits can only be used if the trait is implemented and in scope
  = note: the following traits define an item `ty`, perhaps you need to implement one of them:
          candidate #1: `narrative::step::StepArg`
          candidate #2: `narrative::story::StoryConst`
  = note: this error originates in the attribute macro `narrative::story` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no method named `expr` found for enum `mod_TestStory::args::use_non_local_type` in the current scope
 --> tests/compile-fail/non-local-type.rs:4:1
  |
4 | #[narrative::story("Test non-local type error")]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  | |
  | method not found in `mod_TestStory::args::use_non_local_type`
  | method `expr` not found for this enum
  |
  = help: items from traits can only be used if the trait is implemented and in scope
  = note: the following traits define an item `expr`, perhaps you need to implement one of them:
          candidate #1: `narrative::step::StepArg`
          candidate #2: `narrative::story::StoryConst`
  = note: this error originates in the attribute macro `narrative::story` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no method named `to_dyn` found for enum `mod_TestStory::args::use_non_local_type` in the current scope
 --> tests/compile-fail/non-local-type.rs:4:1
  |
4 | #[narrative::story("Test non-local type error")]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  | |
  | method not found in `mod_TestStory::args::use_non_local_type`
  | method `to_dyn` not found for this enum
  |
  = note: this error originates in the attribute macro `narrative::story` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no method named `name` found for reference `&mod_TestStory::args::use_non_local_type` in the current scope
 --> tests/compile-fail/non-local-type.rs:4:1
  |
4 | #[narrative::story("Test non-local type error")]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ method not found in `&mod_TestStory::args::use_non_local_type`
  |
  = help: items from traits can only be used if the trait is implemented and in scope
  = note: the following traits define an item `name`, perhaps you need to implement one of them:
          candidate #1: `narrative::step::StepArg`
          candidate #2: `narrative::story::StoryConst`
  = note: this error originates in the attribute macro `narrative::story` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no method named `ty` found for reference `&mod_TestStory::args::use_non_local_type` in the current scope
 --> tests/compile-fail/non-local-type.rs:4:1
  |
4 | #[narrative::story("Test non-local type error")]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ method not found in `&mod_TestStory::args::use_non_local_type`
  |
  = help: items from traits can only be used if the trait is implemented and in scope
  = note: the following traits define an item `ty`, perhaps you need to implement one of them:
          candidate #1: `narrative::step::StepArg`
          candidate #2: `narrative::story::StoryConst`
  = note: this error originates in the attribute macro `narrative::story` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no method named `expr` found for reference `&mod_TestStory::args::use_non_local_type` in the current scope
 --> tests/compile-fail/non-local-type.rs:4:1
  |
4 | #[narrative::story("Test non-local type error")]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ method not found in `&mod_TestStory::args::use_non_local_type`
  |
  = help: items from traits can only be used if the trait is implemented and in scope
  = note: the following traits define an item `expr`, perhaps you need to implement one of them:
          candidate #1: `narrative::step::StepArg`
          candidate #2: `narrative::story::StoryConst`
  = note: this error originates in the attribute macro `narrative::story` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no method named `value` found for reference `&mod_TestStory::args::use_non_local_type` in the current scope
 --> tests/compile-fail/non-local-type.rs:4:1
  |
4 | #[narrative::story("Test non-local type error")]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ method not found in `&mod_TestStory::args::use_non_local_type`
  |
  = help: items from traits can only be used if the trait is implemented and in scope
  = note: the following traits define an item `value`, perhaps you need to implement one of them:
          candidate #1: `narrative::step::StepArg`
          candidate #2: `narrative::story::StoryConst`
  = note: this error originates in the attribute macro `narrative::story` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the type `NotLocalType` cannot be used in this story
 --> tests/compile-fail/non-local-type.rs:7:39
  |
7 |     fn use_non_local_type(&self, arg: NotLocalType);
  |                                       ^^^^^^^^^^^^ this type is not allowed in stories
  |
  = help: the trait `LocalType` is not implemented for `NotLocalType`
  = note: only types from the standard library or types defined within the story are allowed
note: required by a bound in `assert_local_type`
 --> tests/compile-fail/non-local-type.rs:4:1
  |
4 | #[narrative::story("Test non-local type error")]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `assert_local_type`
  = note: this error originates in the attribute macro `narrative::story` (in Nightly builds, run with -Z macro-backtrace for more info)
